aplib.net-demo/Assets/Scripts/Player/ResetRigidbody.cs

```diff
@@ -0,0 +1,29 @@
+using UnityEngine;
+
+/// <summary>
+/// Script to reset the position and velocity of a Rigidbody to its initial state.
+/// Requires the object to have a Rigidbody component.
+/// </summary>
+public class ResetRigidbody : MonoBehaviour
+{
+    Rigidbody _rigidbodyToRespawn;
+    private Vector3 _respawnPoint;
```

Comment:
I would recommend linking this to an empty object which you can place in space, which then gets added to this script in a `SerializeField`, giving some more control to developers easily with not much of a development time trade-off.